[Bug]: Error when applying a referral code without a team

### Problem encountered

Error when applying a referral code without a team.

There is an event to state tthat the referral code has been applied successfully , BUT there is also an event stating that the transaction has failed.

**BUS_EVENT_TYPE_TRANSACTION_RESULT Event:**

 `{"id":"121-4","block":"43E25BD6D729DFABA6D71747677A77668EDBB694C8ACFCD376ED6349F216A9A9","type":"BUS_EVENT_TYPE_TRANSACTION_RESULT","transactionResult":{"partyId":"f8bec23cfa1c20128120299765842d5bfcd28f1b66ca1bb27805e7fe7ec4d201","hash":"6658a44772199a12492c9b0ceff2ffb18f0f9b6d5840dec171892c494d1ed98b","applyReferralCode":{"id":"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5"},"failure":{"error":"no team matches ID \"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5\""}},"version":1,"chainId":"testnet-001","txHash":"6658A44772199A12492C9B0CEFF2FFB18F0F9B6D5840DEC171892C494D1ED98B"}`

**BUS_EVENT_TYPE_REFEREE_JOINED_REFERRAL_SET Event:**
   
`['{"id":"121-3","block":"43E25BD6D729DFABA6D71747677A77668EDBB694C8ACFCD376ED6349F216A9A9","type":"BUS_EVENT_TYPE_REFEREE_JOINED_REFERRAL_SET","refereeJoinedReferralSet":{"setId":"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5","referee":"f8bec23cfa1c20128120299765842d5bfcd28f1b66ca1bb27805e7fe7ec4d201","joinedAt":"1694597752968465944","atEpoch":"12"},"version":1,"chainId":"testnet-001","txHash":"6658A44772199A12492C9B0CEFF2FFB18F0F9B6D5840DEC171892C494D1ED98B"}\n']`

### Observed behaviour

Error when applying a referral code without a team.


### Expected behaviour

No error in the event BUS_EVENT_TYPE_TRANSACTION_RESULT

### Steps to reproduce

```bash
Run the below system test:

System Test Name - test_apply_referral_code
System Test Branch - feat/teams-tests

Additional Info:
Create a referral code without a team
Appy that referral code to see the error event.
```


### Software version

vega - develop

### Failing test

test_apply_referral_code

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/8215/testReport/junit/tests.teams/test_team_administration/test_apply_referral_code/

### Configuration used

_No response_

### Relevant log output

```shell
AssertionError: Tx result states failure {"id":"121-4","block":"43E25BD6D729DFABA6D71747677A77668EDBB694C8ACFCD376ED6349F216A9A9","type":"BUS_EVENT_TYPE_TRANSACTION_RESULT","transactionResult":{"partyId":"f8bec23cfa1c20128120299765842d5bfcd28f1b66ca1bb27805e7fe7ec4d201","hash":"6658a44772199a12492c9b0ceff2ffb18f0f9b6d5840dec171892c494d1ed98b","applyReferralCode":{"id":"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5"},"failure":{"error":"no team matches ID \"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5\""}},"version":1,"chainId":"testnet-001","txHash":"6658A44772199A12492C9B0CEFF2FFB18F0F9B6D5840DEC171892C494D1ED98B"}
   BUT a successful apply referral code event is also sent ['{"id":"121-3","block":"43E25BD6D729DFABA6D71747677A77668EDBB694C8ACFCD376ED6349F216A9A9","type":"BUS_EVENT_TYPE_REFEREE_JOINED_REFERRAL_SET","refereeJoinedReferralSet":{"setId":"29009d16d0070a57887a4b58529e6176a098f6553060df96fb9263bf751ac6a5","referee":"f8bec23cfa1c20128120299765842d5bfcd28f1b66ca1bb27805e7fe7ec4d201","joinedAt":"1694597752968465944","atEpoch":"12"},"version":1,"chainId":"testnet-001","txHash":"6658A44772199A12492C9B0CEFF2FFB18F0F9B6D5840DEC171892C494D1ED98B"}\n']
assert False
```

--- a/core/processor/abci.go
+++ b/core/processor/abci.go
@@ -29,6 +29,7 @@ import (
 
 	"code.vegaprotocol.io/vega/core/referral"
 	"code.vegaprotocol.io/vega/core/snapshot"
+	"code.vegaprotocol.io/vega/core/teams"
 	protoapi "code.vegaprotocol.io/vega/protos/vega/api/v1"
 	"go.uber.org/zap"
 
@@ -2262,10 +2263,17 @@ func (app *App) ApplyReferralCode(ctx context.Context, tx abci.Tx) error {
 
 	// it's OK to switch team if the party was already a referrer / referee
 	if err != nil &&
-		!errors.Is(err, referral.ErrIsAlreadyAReferee(partyID)) &&
-		!errors.Is(err, referral.ErrIsAlreadyAReferrer(partyID)) {
+		err.Error() != referral.ErrIsAlreadyAReferee(partyID).Error() &&
+		err.Error() != referral.ErrIsAlreadyAReferrer(partyID).Error() {
 		return fmt.Errorf("could not apply the referral code: %w", err)
 	}
 
-	return app.teamsEngine.JoinTeam(ctx, partyID, params)
+	teamID := types.TeamID(params.Id)
+	err = app.teamsEngine.JoinTeam(ctx, partyID, params)
+	// this is ok as well, as not all referral sets are teams as well
+	if err != nil && err.Error() == teams.ErrNoTeamMatchesID(teamID).Error() {
+		return fmt.Errorf("couldn't join team: %w", err)
+	}
+
+	return nil
 }

